src/js/components/toolbar/MarkDropPlace.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {State} from '../../store';
+import {MarkDraggingStateRecord} from '../../store/factory/Inspector';
+import { LyraMarkType, Mark } from '../../store/factory/Mark';
+import { GroupRecord } from '../../store/factory/marks/Group';
+import {addMark, addGroup } from '../../actions/markActions';
+import {removePlaceHolder} from '../../actions/layoutActions';
+import {getClosestGroupId} from '../../util/hierarchy';
+import { LayoutRecord} from '../../store/factory/Layout';
+
+interface StateProps {
+  dragging: MarkDraggingStateRecord;
+  sceneId: number;
+  layout: LayoutRecord;
+}
+
+interface OwnProps {
+  layoutId: number;
+  placeholderId: number;
+  top: number;
+  left: number;
+  width: number;
+  height: number;
+}
+interface DispatchProps {
+  addMark: (type: LyraMarkType, parentId: number) => void;
+  addGroup: (sceneId: number) => void;
+  removePlaceHolder: () => void;
+}
+
+function mapStateToProps(state: State,  ownProps: OwnProps): StateProps {
+  const draggingRecord = state.getIn(['inspector', 'dragging']);
+  const isMarkDrag = draggingRecord && (draggingRecord as MarkDraggingStateRecord).mark;
+  let sceneId;
+  state.getIn(['vis', 'present', 'marks']).filter(mark => mark.name == "Scene").forEach(scene => sceneId = scene._id);
```

Comment:
This was my work around instead of using getInVis but there is probably a cleaner and clearer way to do this?

Reply:
`state.getIn(['vis', 'present', 'scene', '_id'])`